src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -63,6 +63,9 @@
     /** SQL statement with expanded parameter tokens */
     private String preparedSQL;
 
+    /** True if this execute has been called for this statement at least once */
+    private boolean bExecutedAtLeastOnce = false;
```

Comment:
Nit: you don't need the 'b' before the name of the variable. this can be changed to `executedAtLeastOnce`